src/session.ts

```diff
@@ -357,6 +365,76 @@ export class UserSessionHandler extends BaseSessionHandler {
   }
 }
 
+export class OAuthSessionHandler extends BaseSessionHandler {
+  private readonly _oauthTokenResponse: OAuthTokenResponse
+
+  constructor (connection: MetricsConnection, oauthTokenResponse: OAuthTokenResponse) {
+    super(connection, oauthTokenResponse, false)
+
+    this._oauthTokenResponse = oauthTokenResponse
+  }
+
+  get decodedAccessToken (): any {
+    return DecodeJWT(this.accessToken) as OAuthToken
+  }
+
+  get accessToken () {
+    return this._oauthTokenResponse.accessToken
+  }
+
+  get refreshToken () {
+    return this._oauthTokenResponse.refreshToken
+  }
+
+  get expiresIn () {
+    return this._oauthTokenResponse.expiresIn
+  }
+
+  async logout () {
+    const authParams = { authorization: this.refreshToken ?? this.accessToken }
+    await this._connection.action('auth:logout', authParams)
+    this.close()
+  }
+}
+
+export class OAuthSession {
```

Comment:
Creating a completely new `OAuthSession` class is going to be expensive to maintain because you will have to re-implement every action. You might be better of using the `UserSessionBase` like the `FacilityUserSession`.

Also, you need some way of differentiating the types of `OAuthSession`. Right now there is only a `user` type, but we know that we will likely have a `facility` type before long, so start differentiating between a `UserOAuthSession` and a `FacilityOAuthSession`.